test/tstDistributedSearchTree.cpp

```diff
@@ -374,7 +374,9 @@ BOOST_AUTO_TEST_CASE_TEMPLATE( boost_comparison, DeviceType,
     auto cloud = make_random_cloud( Lx, Ly, Lz, n, 0 );
     auto queries = make_random_cloud( Lx, Ly, Lz, n, 1234 );
 
-    int const local_n = n / comm_size;
+    // The formula is a bit complicated but it does not require n be divisible
+    // by comm_size
+    int const local_n = ( n + comm_size - 1 - comm_rank ) / comm_size;
```

Comment:
Would you post a reference to the place in the code below that would break if `n` is not a multiple of `comm_size`?

Reply:
I have not got a proper trace, but for example the following line will break:
```
for ( int i = 0; i < n; ++i )                                                                                                                                                                                 
{                                                                                                                                                                                                             
  if ( i % comm_size == comm_rank )                                                                                                                                                                         
  {   
>>>  bounding_boxes_host[i / comm_size] = {{{x, y, z}}, {{x, y, z}}};                                                                                                                                      
  }
```
n = 100, comm_size = 6 => bounding_boxes size is 16, but here some will try to access with index 16.

Reply:
Thx